### PR TITLE
Feature/diagnostic plots

### PR DESCRIPTION
**Pull request by jojal5**

### CODE REVIEW

File: src/validationplots/plots.jl [hunk 1]
```diff
@@ -18,12 +37,68 @@ end
 """
 # TODO : desc
 """
-function probplot(fm::MaximumLikelihoodEVA)::Plot
+function probplot_data(fm::BayesianEVA)::DataFrame
```
Comment (MarilouF) [inline src/validationplots/plots.jl]: `probplot_data` is the only plot data function that does not include a function for the `pwmEVA` structure. I think it should be added.

File: src/validationplots/plots.jl [hunk 2]
```diff
@@ -18,12 +37,68 @@ end
...
-        Guide.xlabel("Model"), Guide.ylabel("Empirical"), Guide.title("Probability Plot"))
+        Guide.xlabel("Model"), Guide.ylabel("Empirical"), Guide.title(plotTitle))
+
+end
+
+
+"""
+# TODO : desc
+"""
+function qqplot_data(fm::pwmEVA)::DataFrame
```
Comment (MarilouF) [inline src/validationplots/plots.jl]: Here I am being picky and you can ignore this if you don't think it necessary! 😄 
Every time we have functions related to the `fittedEVA` structures, the order is as follow: `bayesianEVA`, `maximumlikelihoodEVA` and then `pwmEVA` (alphabetical). The data functions here do not follow this order.

File: src/validationplots/plots.jl [hunk 3]
```diff
@@ -1,13 +1,32 @@
+"""
+# TODO : desc
+"""
+function probplot(fm::fittedEVA)::Plot
```
Comment (MarilouF) [inline src/validationplots/plots.jl:4]: This is more of a general comment. Now that we have so many functions, I believe we should rethink our structure. The one I initially thought of is not suitable anymore. 

Maybe we could split it this way:
plots.jl (contains functions that draw the plots)
plots_data.jl (contains non-standardized data functions)
plots_std_data.jl (contains standardized data functions)

We could also think of separating the functions based on the graphs (group together everything related to probability plots, qq plots, etc.)

File: src/validationplots/plots.jl [hunk 4]
```diff
@@ -122,32 +310,57 @@ end
...
     f1 = probplot(fm)
     f2 = qqplot(fm)
-    f3 = returnlevelplot(fm)
-    f4 = histplot(fm)
+    f3 = histplot(fm)
+
+    if getcovariatenumber(fm.model) > 0
+        f4 = plot()
+    else
+        f4 = histplot(fm)
```
Comment (MarilouF) [inline src/validationplots/plots.jl]:
```suggestion
        f4 = returnlevelplot(fm)
```

File: src/validationplots/plots.jl [hunk 5]
```diff
@@ -51,12 +126,66 @@ end
...
+end
+
+
+"""
+# TODO : desc
+"""
+function returnlevelplot(fm::fittedEVA)::Plot
+
+    if getcovariatenumber(fm.model) > 0
+        @warn "this graphic is not optimized for non-stationary model; plot ignored."
```
Comment (MarilouF) [inline src/validationplots/plots.jl:45]: I feel like this is not consistent with the other functions. In the data functions, when the graph is not optimized for the type of model, we `@info`and then still return the values. It is also true for the `returnlevelplot_data` function. I feel like we sould simply `@info` and still return the plot here for consistency. 
Alternatively, we could `@warn` and not return the data in the data functions.

File: src/validationplots/tests_to_integrate.jl [hunk 6]
```diff
@@ -0,0 +1,95 @@
+using Extremes, DataFrames, Gadfly, Test, Distributions
```
Comment (MarilouF) [inline src/validationplots/tests_to_integrate.jl:1]: Do you want me to integrate the tests before we merge?

**Review by MarilouF — CHANGES_REQUESTED** [review]

File: src/validationplots/plots.jl (hunk 1 above)
Comment (jojal5) [reply src/validationplots/plots.jl]: Done

File: src/validationplots/plots.jl (hunk 2 above)
Comment (jojal5) [reply src/validationplots/plots.jl]: Done

File: src/validationplots/plots.jl (hunk 5 above)
Comment (jojal5) [reply src/validationplots/plots.jl:45]: The return level plot makes no sense in the non-stationary context. I would avoid to show it. To make it consistent, changes will be made in returnlevelplot_data

File: src/validationplots/tests_to_integrate.jl (hunk 6 above)
Comment (jojal5) [reply src/validationplots/tests_to_integrate.jl:1]: Not necessarily you :) These tests may be implemented in the near future.

File: src/validationplots/plots.jl (hunk 5 above)
Comment (MarilouF) [reply src/validationplots/plots.jl:45]: Oh ok thanks!

File: src/validationplots/plots.jl (hunk 3 above)
Comment (jojal5) [reply src/validationplots/plots.jl:4]: Done

**Review by MarilouF — APPROVED** [review]

**Pull request merged**